Lib/mimetypes.py

```diff
@@ -94,8 +94,8 @@ def add_type(self, type, ext, strict=True):
         if ext not in exts:
             exts.append(ext)
 
-    def guess_type(self, url, strict=True):
-        """Guess the type of a file based on its URL.
+    def guess_type(self, url_or_path, strict=True):
+        """Guess the type of a file which can either be a url or an os.PathLike object.
```

Comment:
Can you keep "URL" uppercase? Also, this line _seems_ a bit long (though Github doesn't display the line length).